File: src/models/OperacionTalHorizontal.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/sequelize');

const OperacionTalHorizontal = sequelize.define('OperacionTalHorizontal', {
    id: { type: DataTypes.INTEGER, primaryKey: true, autoIncrement: true },
    fecha: DataTypes.STRING,
    turno: DataTypes.STRING,
    seccion: DataTypes.STRING,
    operador: DataTypes.STRING,
    jefe_guardia: DataTypes.STRING,
    equipo: DataTypes.STRING,
    n_equipo: DataTypes.STRING,
    modelo_equipo: DataTypes.STRING,
    registros: DataTypes.TEXT,
    horometros: DataTypes.TEXT,
    condiciones_equipo: DataTypes.TEXT,
    check_list: DataTypes.TEXT,
    control_llantas: DataTypes.TEXT,
    estado: { type: DataTypes.STRING, defaultValue: 'activo' },
    envio: { type: DataTypes.INTEGER, defaultValue: 0 },
    revisado: { 
        type: DataTypes.INTEGER, 
        defaultValue: 0 
    },
 
    aprobacion: { 
  type: DataTypes.INTEGER, 
  defaultValue: 0 
},
   observaciones_jefe: {
      type: DataTypes.JSON,
      allowNull: true,
    },

    observaciones_jefe2: {
      type: DataTypes.JSON,
      allowNull: true,
    },
    observaciones_jefe3: {
      type: DataTypes.JSON,
      allowNull: true,
    },
}, {
    tableName: 'Operacion_tal_horizontal',
    timestamps: false
});

module.exports = OperacionTalHorizontal;

File: dotnet/src/Seminco.Domain/OperacionesV2/OperacionTalHorizontal.cs
namespace Seminco.Domain.OperacionesV2;

public sealed class OperacionTalHorizontal
{
    public int Id { get; set; }
    public DateTime Fecha { get; set; }
    public string? Turno { get; set; }
    public string? Operador { get; set; }
    public string? JefeGuardia { get; set; }

    public int? EquipoId { get; set; }
    public string? EquipoNombre { get; set; }
    public string? NEquipo { get; set; }

    public string? Seccion { get; set; }
    public string? ModeloEquipo { get; set; }

    public string Estado { get; set; } = "activo";
    public int Envio { get; set; }
    public int Revisado { get; set; }
    public int Aprobacion { get; set; }

    public string? ObservacionesJefe { get; set; }
    public string? ObservacionesJefe2 { get; set; }
    public string? ObservacionesJefe3 { get; set; }

    public string? PayloadOriginal { get; set; }
    public string? PayloadVersion { get; set; }
    public string? ExternalSyncId { get; set; }
    public string? DeviceId { get; set; }

    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    public OperacionTalHorizontalCondicionEquipo? CondicionEquipo { get; set; }
    public List<OperacionTalHorizontalHorometro> Horometros { get; set; } = [];
    public List<OperacionTalHorizontalChecklistRespuesta> ChecklistRespuestas { get; set; } = [];
    public List<OperacionTalHorizontalControlLlanta> ControlLlantas { get; set; } = [];
    public List<OperacionTalHorizontalRegistro> Registros { get; set; } = [];
}